vdom/core.py

```diff
@@ -122,14 +129,42 @@ def to_dict(self):
     def to_json(self):
         return json.dumps(self.to_dict())
 
+    def to_html(self):
+        return self._repr_html_()
+
     def json_contents(self):
         warnings.warn('VDOM.json_contents method is deprecated, use to_json instead')
         return self.to_json()
 
+    def _repr_html_(self):
+        """
+        Return HTML representation of VDOM object.
+
+        HTML escaping is performed wherever necessary.
+        """
+        # Use StringIO to avoid a large number of memory allocations with string concat
+        with io.StringIO() as out:
+            out.write('<{tag}'.format(tag=escape(self.tag_name)))
+
+            for k, v in self.attributes.items():
+                # Important values are in double quotes - cgi.escape only escapes double quotes, not single quotes!
+                out.write(' {key}="{value}"'.format(key=escape(k), value=escape(v)))
+            out.write('>')
+
+            for c in self.children:
+                if isinstance(c, str):
+                    out.write(escape(c))
+                else:
+                    out.write(c._repr_html_())
+
+            out.write('</{tag}>'.format(tag=escape(self.tag_name)))
+
+            return out.getvalue()
+
     def _repr_mimebundle_(self, include, exclude, **kwargs):
         return {
             'application/vdom.v1+json': self.to_dict(),
-            'text/plain': '<{tagName} />'.format(tagName=self.tag_name)
+            'text/plain': self.to_html()
```

Comment:
Also, since I don't want this to block the PR whatsoever, I'd take out the `to_html` from occurring by default and defer the configurability aspect for another PR.

Reply:
I'm on board with that.